sympy/solvers/solvers.py

```diff
@@ -375,6 +375,60 @@ def checksol(f, symbol, sol=None, **flags):
     # returns None if it can't conclude
     # TODO: improve solution testing
 
+def _is_eq_bool(eq):
+    """Returns true if Eq or Ne has True or False as an argument"""
+    if isinstance(eq, (Equality, Unequality)):
+        return any(side in (S.false, S.true) for side in eq.args)
+
+def _invert_eq_bool(fi):
+    """Returns the expression when one argument is a bool and the other an expression in an Eq or Ne"""
+    if isinstance(fi, (Equality, Unequality)):
+        lhs, rhs = fi.args
+        unequal = S.true if isinstance(fi, Unequality) else S.false
+        if lhs in (S.false, S.true) and rhs.is_Relational:
+            return ~rhs if lhs is unequal else rhs
+        if rhs in (S.false, S.true) and lhs.is_Relational:
+            return ~lhs if rhs is unequal else lhs
+    return fi
+
+def _is_relational(fi):
+    return fi.is_Relational and not isinstance(fi, Equality)
```

Comment:
I'm not sure that moving a trivial function like this so far from where it is used helps.